### ISSUE TITLE
Unasked-for scaling in tweened rotations

### ISSUE THREAD

**Issue by anoopalexwork** [tahoma2d/tahoma2d#1327]
### Bug Description

Noticed a weird glitch in tweened rotation. Whenever i change the rotation of an object using transform box of Select tool in another frame, if I tween between the 2 frames, only scaling takes places till very near midpoint of tween, when quick rotation takes place for a few frames, then scaling back up only till the end of tween when full original scale size is regained. The amount of lowest scale at midpoint is relative to the length of the tween. Happens whether I rotate clockwise or counter clockwise. Also though the smaller objects can be selected in each frame, the H V Scale values in Select Tool are always at 100%.

### Steps to reproduce

- Pick an object in a frame
- duplicate it
- Rotate duplicate using transform box of Select tool. Doesn't matter if pivot point is changed or not.
- Insert a number of frames between duplicate and original
- Select inserted frames, original frame and duplicate frame
- Perform tween
- Irregularly rotated scaled object is present in tween frames

### Relevant logs/screenshots

Here is a short video of an example tween rotation and the individual frames created

https://github.com/tahoma2d/tahoma2d/assets/76545170/b2e2a19a-b98c-4ffb-aac1-4014922bdf8a

### Software Version

1.4

### Operating System

Window 10

### Graphics Card

NVIDIA GeForce GTX 1060 6GB 

### Drawing Tablet

Huion Inspiroy Dial

**Comment by beeheemooth** [comment]
Auto-tweening in such cases is done in conjunction with key animation. If your hand is moving, rotating or scaling, it's best to do this using the Animate Tool. Auto-tweening is designed to transform the form itself. You can do both key animation and Auto-tweening at the same time to get the desired result.

**Comment by anoopalexwork** [comment]
> Auto-tweening in such cases is done in conjunction with key animation. If your hand is moving, rotating or scaling, it's best to do this using the Animate Tool. Auto-tweening is designed to transform the form itself. You can do both key animation and Auto-tweening at the same time to get the desired result.

But Animate affects the whole level. If I wish to rotate only some of the vector objects, then my only option is to place those objects on individual levels. Is the issue an unexpected glitch in the app or a known behavior when autotweening?

**Comment by beeheemooth** [comment]
Yes, the Animate Tool affects the entire column. If you draw your character in one level and want to use auto-tweening for some part, then it is advisable to temporarily place this part on a separate level and make the animation also using keys. The point is that key animation and auto-tweening are intended for different purposes and complement each other. This video shows the result of using only auto-tweening, as well as auto-tweening along with key animation.

![truba](https://github.com/tahoma2d/tahoma2d/assets/25724133/cca9afa9-b9da-49ca-9b3e-34f7e52a341a)

**Comment by manongjohn** [comment]
It is interesting that it isn't rotating the lines except towards the middle frames.

Is that hand multiple lines or 1 line. It appears the line(s) in that hand grouped. Not sure, but may not like that it is grouped.

I believe there might also be a tweening issues with copy pasting since the direction of the drawn vector points maybe reversed and drawn direction matters in tweening. Maybe instead of copy pasting, duplicate the 1st frame, renumber it to the end and then rotate and try tweeening again? Also can use the Vector Tweening Panel to make sure the direction of the drawn lines is the same.

**Comment by beeheemooth** [comment]
I think I understand what @anoopalexwork  means. I reproduced the problem by drawing a similar hand. It really consists of one closed and grouped vector.
![ezgif-4-c767d48778](https://github.com/tahoma2d/tahoma2d/assets/25724133/605d0aef-fb01-4544-9c6c-ccf5ea530b72)

**Comment by beeheemooth** [comment]
Flip Vertical + Flip Horizontal gives the same result as a 180 degree rotation. We were deceived, hehe!

**Comment by manongjohn** [comment]
Strange. I just attempted to duplicate the tweened rotation and mine rotated correctly. 🤔

**Comment by beeheemooth** [comment]
@manongjohn Try drawing a hand with the "Auto Close" option enabled.

**Comment by anoopalexwork** [comment]
> 
> Is that hand multiple lines or 1 line. It appears the line(s) in that hand grouped. Not sure, but may not like that it is grouped.

It is a single closed polyline. I tried duplicating a brush stroke that was closed too and duplicating but same thing happens.
> 
> I believe there might also be a tweening issues with copy pasting since the direction of the drawn vector points maybe reversed and drawn direction matters in tweening. Maybe instead of copy pasting, duplicate the 1st frame, renumber it to the end and then rotate and try tweeening again? Also can use the Vector Tweening Panel to make sure the direction of the drawn lines is the same.
Can I see vector guidelines when I have already duplicated the only stroke in the frame? Also I've tried rotation CW and CCW but effect is same.

**Comment by anoopalexwork** [comment]
Was anybody able to recreate the bug? I see this bug even if I start a new level or new scene. I even installed T2D into a different directory, tested that in a different scene there, creating a new object and rotating and the exact same thing was noticed there! Even tried transforming using the Rotation text box, entered 90 degrees but still same effect.

> I think I understand what @anoopalexwork means. I reproduced the problem by drawing a similar hand. It really consists of one closed and grouped vector. ![ezgif-4-c767d48778](https://private-user-images.githubusercontent.com/25724133/293469975-605d0aef-fb01-4544-9c6c-ccf5ea530b72.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM5NzM0NzIsIm5iZiI6MTcwMzk3MzE3MiwicGF0aCI6Ii8yNTcyNDEzMy8yOTM0Njk5NzUtNjA1ZDBhZWYtZmIwMS00NTQ0LTljNmMtY2NmNWVhNTMwYjcyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyMzEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjMwVDIxNTI1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwNTBlNTM1NjRjNjQ1YmFiYzJjMzUyZTZjOGUwYzhkYjJmNjk2ODViYzMzNzc0MTI3YzI3MTA3MTAwZGY5ODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TTbVTLuOxcXI9ONToQ-wdCQm0gqKf9inGIN7hao2lsk) [ ![ezgif-4-c767d48778](https://private-user-images.githubusercontent.com/25724133/293469975-605d0aef-fb01-4544-9c6c-ccf5ea530b72.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM5NzM0NzIsIm5iZiI6MTcwMzk3MzE3MiwicGF0aCI6Ii8yNTcyNDEzMy8yOTM0Njk5NzUtNjA1ZDBhZWYtZmIwMS00NTQ0LTljNmMtY2NmNWVhNTMwYjcyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyMzEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjMwVDIxNTI1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwNTBlNTM1NjRjNjQ1YmFiYzJjMzUyZTZjOGUwYzhkYjJmNjk2ODViYzMzNzc0MTI3YzI3MTA3MTAwZGY5ODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TTbVTLuOxcXI9ONToQ-wdCQm0gqKf9inGIN7hao2lsk) ](https://private-user-images.githubusercontent.com/25724133/293469975-605d0aef-fb01-4544-9c6c-ccf5ea530b72.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM5NzM0NzIsIm5iZiI6MTcwMzk3MzE3MiwicGF0aCI6Ii8yNTcyNDEzMy8yOTM0Njk5NzUtNjA1ZDBhZWYtZmIwMS00NTQ0LTljNmMtY2NmNWVhNTMwYjcyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyMzEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjMwVDIxNTI1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwNTBlNTM1NjRjNjQ1YmFiYzJjMzUyZTZjOGUwYzhkYjJmNjk2ODViYzMzNzc0MTI3YzI3MTA3MTAwZGY5ODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TTbVTLuOxcXI9ONToQ-wdCQm0gqKf9inGIN7hao2lsk) [ ](https://private-user-images.githubusercontent.com/25724133/293469975-605d0aef-fb01-4544-9c6c-ccf5ea530b72.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM5NzM0NzIsIm5iZiI6MTcwMzk3MzE3MiwicGF0aCI6Ii8yNTcyNDEzMy8yOTM0Njk5NzUtNjA1ZDBhZWYtZmIwMS00NTQ0LTljNmMtY2NmNWVhNTMwYjcyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyMzEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjMwVDIxNTI1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwNTBlNTM1NjRjNjQ1YmFiYzJjMzUyZTZjOGUwYzhkYjJmNjk2ODViYzMzNzc0MTI3YzI3MTA3MTAwZGY5ODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TTbVTLuOxcXI9ONToQ-wdCQm0gqKf9inGIN7hao2lsk)

Ok. You were able to reproduce it! Thats a first step at least. Any idea why?

**Comment by manongjohn** [comment]
It appears to be due to lines that are auto closed or already closed shapes, like polygons and rectangles.  Interestingly, if you do less than 180° (i.e 135°), you will see more rotation frames, though it does scale down and back up again.

For non-auto closed shapes, there is a difference when rotating 180° via the rotation buttons vs manual rotation.

**Comment by anoopalexwork** [comment]
> It appears to be due to lines that are auto closed or already closed shapes, like polygons and rectangles. Interestingly, if you do less than 180° (i.e 135°), you will see more rotation frames, though it does scale down and back up again.

Yeah, youre right. Thanks for picking that up manongjohn. I didn't catch that since all my shapes were made with closed polyline. I've trying to learn more about vector editing so I avoid brush for character drawing and switch to polyline for cleaner lines. Fills also get problematic when editing unclosed objects.
 
I also tried end-to-end taping an unclosed polyline and rotating it but didn't work.
